spanner/transaction_test.go

```diff
@@ -504,6 +504,101 @@ func TestReadWriteTransaction_PrecommitToken(t *testing.T) {
 	}
 }
 
+func TestCommitWithMultiplexedSessionRetry(t *testing.T) {
+	ctx := context.Background()
+	server, client, teardown := setupMockedTestServerWithConfig(t, ClientConfig{
+		DisableNativeMetrics: true,
+		SessionPoolConfig: SessionPoolConfig{
+			MinOpened:                     1,
+			MaxOpened:                     1,
+			enableMultiplexSession:        true,
+			enableMultiplexedSessionForRW: true,
+		},
+	})
+	defer teardown()
+
+	// newCommitResponseWithPrecommitToken creates a simulated response with a PrecommitToken
+	newCommitResponseWithPrecommitToken := func() *sppb.CommitResponse {
+		precommitToken := &sppb.MultiplexedSessionPrecommitToken{
+			PrecommitToken: []byte("commit-retry-precommit-token"),
+		}
+
+		// Create a CommitResponse with the PrecommitToken
+		return &sppb.CommitResponse{
+			MultiplexedSessionRetry: &sppb.CommitResponse_PrecommitToken{PrecommitToken: precommitToken},
+		}
+	}
+
+	// Simulate a commit response with a MultiplexedSessionRetry
+	server.TestSpanner.PutExecutionTime(MethodCommitTransaction,
+		SimulatedExecutionTime{
+			Responses: []interface{}{newCommitResponseWithPrecommitToken()},
+		})
+
+	_, err := client.ReadWriteTransaction(ctx, func(ctx context.Context, tx *ReadWriteTransaction) error {
+		ms := []*Mutation{
+			Insert("t_foo", []string{"col1", "col2"}, []interface{}{int64(1), int64(2)}),
+			Update("t_foo", []string{"col1", "col2"}, []interface{}{"one", []byte(nil)}),
+		}
+		if err := tx.BufferWrite(ms); err != nil {
+			return err
+		}
+
+		iter := tx.Query(ctx, NewStatement(SelectSingerIDAlbumIDAlbumTitleFromAlbums))
+		defer iter.Stop()
+		for {
+			_, err := iter.Next()
+			if err == iterator.Done {
+				break
+			}
+			if err != nil {
+				return err
+			}
+		}
+
+		if _, err := tx.Update(ctx, Statement{SQL: UpdateBarSetFoo}); err != nil {
+			return err
```

Comment:
Commit retry only happens in cases of DQL+Mutation. Lets remove DML to avoid confusion